Disable handler-tls in end2end tests

menghanl opened the pull request:
Some end2end tests using ServeHTTP failed with DATA RACE:
https://travis-ci.org/grpc/grpc-go/jobs/169080239
https://travis-ci.org/grpc/grpc-go/jobs/169080240
https://travis-ci.org/grpc/grpc-go/jobs/169080236
probably due to the recent change https://github.com/golang/net/commit/3bafa3320efdf0c95d056586aa9abdd05ad72ee1

It seems http2 package is not stable enough.
Disable tests for ServeHTTP in end2end tests until it's stable.

tamird commented:
This seems like a Bad Thing. Paging @bradfitz because disabling tests is not a solution.

bradfitz commented:
Agree. If I introduced a bug the answer is to fix it, not ignore it.

I'll investigate.

bradfitz commented:
Okay, I think I see what's happening. The body Read and Close calls are happening from different goroutines which isn't really expected. 

I'll fix either side or both, and docs and tests.

But not at a computer at the moment. Will look in a few hours after dinner.

bradfitz commented:
Why does your proposed patch do more than just comment out the one bit? Why all the deleted lines?

iamqizhao commented:
@bradfitz Thanks for taking care of this. We will only comment out the failed tests in this PR. They can be revived once your fix is in.

menghanl commented:
The deleted lines were added to disable tests for "handler-tls" in those particular tests. I deleted them because I thought since "handlrEnv" was commented out, we don't need those any more.
I have reverted the change so we will keep those lines.

We need to merge this PR so we can continue to work on other PRs without breaking the test.
And we should revert after the original problem is fixed.

bradfitz commented:
Sent out the fix: https://golang.org/cl/31636
